RasterDE_BY_GK4_UTM32DirInv.py

```diff
@@ -0,0 +1,156 @@
+        self.addParameter(QgsProcessingParameterEnum(self.GRID,
+                                                     'NTv2 Grid',
+                                                     options=[i[0] for i in self.grids],
+                                                     defaultValue=0))
+        self.addParameter(QgsProcessingParameterRasterDestination(self.OUTPUT,
+                                                                  'Output'))
+
+    def getConsoleCommands(self, parameters, context, feedback, executing=True):
+        inLayer = self.parameterAsRasterLayer(parameters, self.INPUT, context)
+        if inLayer is None:
+            raise QgsProcessingException(self.invalidRasterError(parameters, self.INPUT))
+
+        outFile = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
+        self.setOutputValue(self.OUTPUT, outFile)
+
+        direction = self.parameterAsEnum(parameters, self.TRANSF, context)
+        epsg = self.datums[self.parameterAsEnum(parameters, self.CRS, context)][1]
+        grid = self.grids[self.parameterAsEnum(parameters, self.GRID, context)][1]
+
+        found, text = de_transformation(epsg, grid)
+        if not found:
+           raise QgsProcessingException(text)
+
+        arguments = []
+
+        if direction == 0:
+            # Direct transformation
+            arguments.append('-s_srs')
+            arguments.append(text)
+            arguments.append('-t_srs')
+            arguments.append('EPSG:25832')
+        else:
+            # Inverse transformation
+            arguments = ['-s_srs']
+            arguments.append('EPSG:25832')
+            arguments.append('-t_srs')
+            arguments.append(text)
+
+        arguments.append('-multi')
+        arguments.append('-of')
+        arguments.append(QgsRasterFileWriter.driverForExtension(os.path.splitext(outFile)[1]))
+        arguments.append(inLayer.source())
+        arguments.append(outFile)
+
+        gridFile = os.path.join(pluginPath, 'grids', 'BY_KANU.gsb')
+        if not os.path.isfile(gridFile):
+            #TODO (Best way, altough a lot of effort): 1) Display popup which informs the user about the big file size with OK and EXIT Buttons > OnClick OK: Download file with progress bar... 3) save downloaded file and unzip it in the plugins folder 4) proceed with transforming the geodata
+            error_message = 'ERROR GRID FILE NOT FOUND / USER ACTION REQUIRED:\nThe grid file BY_KANU.gsb is to big to be downloaded in the background (3.03 GB), but therefore has a high transformation accuracy of +- 1cm.\n\nFollowing steps are required:\n1. Download the grid from "http://geodaten.bayern.de/oadownload/bvv_internet/kanu/ntv2_bayern.zip"\n2. Unzip the file and rename it to "BY_KANU.gsb"\n3. Move the file to following location: "{}"'.format(gridFile)
+            QgsMessageLog.logMessage(error_message, level=Qgis.Critical)
+            iface.messageBar().pushMessage(error_message, level=Qgis.Critical, duration=30)
```

Comment:
Processing algorithms should not use pop-up dialogs and message bars. The correct way is to raise exception.

Reply:
switched to QgsMessageLog and QgsProcessingException.